### PR TITLE
ARROW-70: Add adapt 'lite' DCHECK macros from Kudu as also used in Parquet

### PR DESCRIPTION
**Pull request by wesm**
Also added a null pointer DCHECK to show that it works. cc @emkornfield

### CODE REVIEW

File: cpp/src/arrow/ipc/adapter.cc [hunk 1]
```diff
@@ -41,6 +42,7 @@ namespace flatbuf = apache::arrow::flatbuf;
 namespace ipc {
 
 static bool IsPrimitive(const DataType* type) {
+  DCHECK(type != nullptr);
```
Comment (emkornfield) [inline cpp/src/arrow/ipc/adapter.cc]: Do you prefer this or `DCHECK(type)`?  For checks like this I think I prefer this without explicit comparison, but its not a strong preference.

**Comment by emkornfield** [comment]
Thank you looks good.

**Comment by wesm** [comment]
+1, thank you.

File: cpp/src/arrow/ipc/adapter.cc (hunk 1 above)
Comment (wesm) [inline cpp/src/arrow/ipc/adapter.cc]: I don't have a strong preference either.

**Pull request closed without merging**